core/commonMain/src/kotlinx/io/internal/LeakDetectingPool.kt

```diff
@@ -0,0 +1,46 @@
+@file:Suppress("FORBIDDEN_IDENTITY_EQUALS")
+
+package kotlinx.io.internal
+
+import kotlinx.io.buffer.*
+import kotlinx.io.pool.*
+
+/**
+ * The pool that tracks every borrowed object and check if it belongs to this pool.
+ *
+ * It also verifies if this pool is empty at close.
+ */
+internal class LeakDetectingPool(private val size: Int = DEFAULT_BUFFER_SIZE) : ObjectPool<Buffer> {
```

Comment:
Looks like it should belong to test module

Reply:
Yep, but we can't share them across test modules